account_journal_lock_date/models/account_move.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import date
+
+from odoo import _, models
+from odoo.tools.misc import format_date
+
+from ..exceptions import JournalLockDateError
+
+
+class AccountMove(models.Model):
+
+    _inherit = "account.move"
+
+    def write(self, values):
+        res = super().write(values)
+        self._check_fiscalyear_lock_date()
+        return res
+
+    def _check_fiscalyear_lock_date(self):
+        res = super()._check_fiscalyear_lock_date()
+        if self.env.context.get("bypass_journal_lock_date"):
+            return res
+        for move in self:
+            if self.user_has_groups("account.group_account_manager"):
+                lock_date = move.journal_id.fiscalyear_lock_date or date.min
+            else:
+                lock_date = max(
+                    move.journal_id.period_lock_date or date.min,
+                    move.journal_id.fiscalyear_lock_date or date.min,
+                )
+            if move.date <= lock_date:
+                lock_date = format_date(self.env, lock_date)
+                if self.user_has_groups("account.group_account_manager"):
+                    message = _(
+                        "You cannot add/modify entries for the journal '%s' "
+                        "prior to and inclusive of the lock date %s"
+                    ) % (move.journal_id.display_name, lock_date)
+                else:
+                    message = _(
+                        "You cannot add/modify entries for the journal '%s' "
+                        "prior to and inclusive of the lock date %s. "
+                        "Check the Journal settings or ask someone "
+                        "with the 'Adviser' role"
+                    ) % (move.journal_id.display_name, lock_date)
+                raise JournalLockDateError(message)
```

Comment:
Module works well, but this exception would give me a full error message with this line at the end rather than the usual UserError-like window. Either I'm missing something on my end or these custom exceptions need to be declared differently in 14.0, anyone know anything about this? I couldn't find other OCA custom errors and replicating base Odoo ones didn't work

Reply:
It might just work if you add the custom exception name to this registry here: https://github.com/odoo/odoo/blob/14.0/addons/web/static/src/js/services/crash_manager.js#L92 in a small javascript override.